compiler/modules/types/list.bal

```diff
@@ -347,18 +349,31 @@ function fixedArraySet(FixedLengthArray members, int setIndex, SemType m) {
         members.initial[setIndex] = m;
         return;
     }
-        
     SemType lastMember = members.initial[initCount - 1]; 
-    foreach int i in initCount ..< setIndex + 1 {
-        members.initial.push(lastMember);
+    if lastMember != m {
+        foreach int i in initCount ..< setIndex + 1 {
+            members.initial.push(lastMember);
+        }
+        members.initial[setIndex] = m;
     }
-    members.initial[setIndex] = m;
+    members.fixedLength = int:max(members.fixedLength, setIndex + 1);
```

Comment:
Is it valid to call this function with setIndex >= members.fixedLength?

Reply:
There are places where we need to go over the `fixedLength` and set some individual member.